src/external_trait_impls/rayon/raw.rs

```diff
@@ -0,0 +1,539 @@
+/// [`HashMap`]: /hashbrown/struct.HashMap.html
+/// [`IntoParallelRefMutIterator`]: https://docs.rs/rayon/1.0/rayon/iter/trait.IntoParallelRefMutIterator.html
+pub struct ParIterMut<'a, K: 'a, V: 'a> {
+    table: &'a mut RawTable<(K, V)>,
+}
+
+impl<'a, K: Sync, V: Send> IntoParallelIterator for &'a mut RawTable<(K, V)> {
+    type Item = (&'a K, &'a mut V);
+    type Iter = ParIterMut<'a, K, V>;
+
+    #[inline]
+    fn into_par_iter(self) -> Self::Iter {
+        ParIterMut { table: self }
+    }
+}
+
+unsafe impl<'a, K: Sync, V: Send> Send for ParIterMut<'a, K, V> {}
+
+impl<'a, K: Sync, V: Send> ParallelIterator for ParIterMut<'a, K, V> {
+    type Item = (&'a K, &'a mut V);
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+    where
+        C: UnindexedConsumer<Self::Item>,
+    {
+        let buckets = unsafe { self.table.iter().iter };
+        let producer = ParIterMutProducer::from(buckets);
+        plumbing::bridge_unindexed(producer, consumer)
+    }
+}
+
+impl<'a, K: fmt::Debug, V: fmt::Debug> fmt::Debug for ParIterMut<'a, K, V> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let iter = unsafe { self.table.iter() };
+        let entries = iter.map(|bucket| unsafe {
+            let (ref k, ref v) = *bucket.as_ref();
+            (k, v)
+        });
+
+        f.debug_list().entries(entries).finish()
+    }
+}
+
+struct ParIterMutProducer<'a, K: 'a, V: 'a> {
+    iter: RawIterRange<(K, V)>,
+    // To ensure invariance with respect to V
+    marker: PhantomData<&'a mut V>,
+}
+
+impl<'a, K, V> From<RawIterRange<(K, V)>> for ParIterMutProducer<'a, K, V> {
+    #[inline]
+    fn from(iter: RawIterRange<(K, V)>) -> Self {
+        Self { iter, marker: PhantomData }
+    }
+}
+
+unsafe impl<'a, K: Sync, V: Send> Send for ParIterMutProducer<'a, K, V> {}
+
+impl<'a, K: Sync, V: Send> UnindexedProducer for ParIterMutProducer<'a, K, V> {
+    type Item = (&'a K, &'a mut V);
+
+    fn split(self) -> (Self, Option<Self>) {
+        unsafe { raw_iter_range_split_and_convert(self.iter) }
+    }
+
+    fn fold_with<F>(self, folder: F) -> F
+    where
+        F: Folder<Self::Item>,
+    {
+        let iter = self.iter.map(|bucket| unsafe {
+            let (ref k, ref mut v) = *bucket.as_mut();
+            (k, v)
+        });
+        folder.consume_iter(iter)
+    }
+}
+
+
+/// Parallel iterator over mutable references to values in a map.
+///
+/// This iterator is created by the [`par_values_mut`] method on [`HashMap`].
+/// See its documentation for more.
+///
+/// [`par_values_mut`]: /hashbrown/struct.HashMap.html#method.par_values_mut
+/// [`HashMap`]: /hashbrown/struct.HashMap.html
+pub struct ParValuesMut<'a, K: 'a, V: 'a> {
+    table: &'a mut RawTable<(K, V)>,
+}
+
+impl<K, V: Send> RawTable<(K, V)> {
+    #[inline]
+    pub fn par_values_mut(&mut self) -> ParValuesMut<K, V> {
+        ParValuesMut { table: self }
+    }
+}
+
+unsafe impl<'a, K, V: Send> Send for ParValuesMut<'a, K, V> {}
```

Comment:
This is not exactly implied by `&'a mut RawTable<(K, V)>`, but it is good enough. Remove this impl.